runtime/resolver.go

```diff
@@ -34,6 +34,8 @@ type Resolver interface {
 	ResolveInteractive(ctx context.Context) (*ResolverResult, error)
 	// ResolveExport resolve data for export (e.g. downloads or reports).
 	ResolveExport(ctx context.Context, w io.Writer, opts *ResolverExportOptions) error
+	// ResolveSchema returns just the schema for the query
+	ResolveSchema(ctx context.Context) (*runtimev1.StructType, error)
```

Comment:
Can we just change `ResolverResult` to include the result schema when calling `ResolveInteractive`?

The `ResolveSchema` implementation anyway runs the full query right now, and I'm not sure we'll be able to avoid running the query to get the schema for some resolvers.

Running the query is probably anyway faster and cheaper than invoking and waiting for the AI, so shouldn't be a bottleneck :)

Reply:
True since the final query would have to be run in the UI for preview this shouldnt be a problem.